R-package/R/xgb.DMatrix.R

```diff
@@ -199,6 +202,35 @@ xgb.DMatrix <- function(
   return(dmat)
 }
 
+#' @title Check whether DMatrix object has a field
+#' @description Checks whether an xgb.DMatrix object has a given field assigned to
+#' it, such as weights, labels, etc.
+#' @param object The DMatrix object to check for the given \code{info} field.
+#' @param info The field to check for presence or absence in \code{object}.
+#' @seealso \link{xgb.DMatrix}, \link{getinfo.xgb.DMatrix}, \link{setinfo.xgb.DMatrix}
+#' @examples
+#' library(xgboost)
+#' x <- matrix(1:10, nrow = 5)
+#' dm <- xgb.DMatrix(x, nthread = 1)
+#'
+#' # 'dm' so far doesn't have any fields set
+#' xgb.DMatrix.hasinfo(dm, "label")
+#'
+#' # Fields can be added after construction
+#' setinfo(dm, "label", 1:5)
+#' xgb.DMatrix.hasinfo(dm, "label")
+#' @export
+xgb.DMatrix.hasinfo <- function(object, info) {
+  if (!inherits(object, "xgb.DMatrix")) {
+    stop("Object is not an 'xgb.DMatrix'.")
+  }
+  if (.Call(XGCheckNullPtr_R, object)) {
+    warning("xgb.DMatrix object is invalid. Must be constructed again.")
```

Comment:
Can it happen?

Reply:
Yes, it can happen when the object is serialized and then de-serialized, for example:
```r
library(xgboost)
x <- matrix(1:10, nrow=5)
dm <- xgb.DMatrix(x)
dm2 <- dm |> serialize(NULL) |> unserialize()
xgb.DMatrix.hasinfo(dm2, "label")
```
```
[1] FALSE
Warning message:
In xgb.DMatrix.hasinfo(dm2, "label") :
  xgb.DMatrix object is invalid. Must be constructed again.
```